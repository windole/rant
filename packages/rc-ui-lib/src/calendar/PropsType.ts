import React, { MouseEvent } from 'react';
import { PopupPosition } from '../popup';
import { BaseTypeProps, TeleportType } from '../utils';

export type CalendarType = 'single' | 'range' | 'multiple';

export type CalendarDayType =
  | ''
  | 'start'
  | 'start-end'
  | 'middle'
  | 'end'
  | 'selected'
  | 'multiple-middle'
  | 'multiple-selected'
  | 'disabled'
  | 'placeholder';

export type CalendarDayItem = {
  date?: Date;
  text?: string | number;
  type?: CalendarDayType;
  topInfo?: string;
  className?: string;
  bottomInfo?: string;
};

export type CalendarDayProps = {
  item: CalendarDayItem;
  color: string;
  index: number;
  offset?: number;
  rowHeight: string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarInstance = {
  reset: (date?: Date | Date[]) => void;
  scrollToDate: (targetDate: Date) => void;
};

export interface CalendarProps extends BaseTypeProps {
  position?: PopupPosition;
  round?: boolean;
  closeOnPopstate?: boolean;
  closeOnClickOverlay?: boolean;
  safeAreaInsetBottom?: boolean;
  teleport?: TeleportType;
  show?: boolean;
  type?: CalendarType;
  title?: React.ReactNode | string;
  subtitle?: React.ReactNode | string;
  color?: string;
  readonly?: boolean;
  poppable?: boolean;
  maxRange?: string | number;
  showMark?: boolean;
  showTitle?: boolean;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  rowHeight?: string | number;
  confirmText?: string;
  rangePrompt?: string;
  lazyRender?: boolean;
  showConfirm?: boolean;
  defaultDate?: Date | Date[] | null;
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showRangePrompt?: boolean;
  confirmDisabledText?: string;
  minDate?: Date;
  maxDate?: Date;
  firstDayOfWeek?: number;
  footer?: React.ReactNode | string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClickSubtitle?: (ev: MouseEvent) => void;
  /** 点击确认后的回调 */
  onConfirm?: (v: Date | Date[]) => void;
  onOverRange?: () => void;
  onSelect?: (v: Date | Date[]) => void;
  onUnselect?: (v: Date | Date[]) => void;
  onMonthShow?: (obj: { date: Date; title: string }) => void;
  onClose?: () => void;
}

export type CalendarMonthProps = {
  date?: Date;
  type?: CalendarType;
  color?: string;
  minDate?: Date;
  maxDate?: Date;
  showMark?: boolean;
  rowHeight?: number | string;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  formatMonthTitle?: (date: Date) => React.ReactNode;
  lazyRender?: boolean;
  currentDate?: Date | Date[];
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showMonthTitle?: boolean;
  firstDayOfWeek?: number;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarMonthInstance = {
  showed?: boolean;
  getTitle: () => string | React.ReactNode;
  getHeight: () => number;
  setVisible: (value?: boolean | undefined) => void;
  scrollToDate: (body: Element, targetDate: Date) => void;
};
