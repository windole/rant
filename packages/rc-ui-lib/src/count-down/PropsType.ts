import React from 'react';
import { CurrentTime } from '../hooks/use-count-down';
import { BaseTypeProps } from '../utils';

export interface CountDownProps extends BaseTypeProps {
  /** 	倒计时时长，单位毫秒 */
  time?: number | string;
  /** 时间格式	 */
  format?: string;
  /**  是否自动开始倒计时 */
  autoStart?: boolean;
  /**  是否开启毫秒级渲染 */
  millisecond?: boolean;
  /**  倒计时结束时触发 */
  onFinish?: () => void;
  /**  是否开启毫秒级渲染 */
  onChange?: (value: CurrentTime) => void;
  renderChildren?: (value: CurrentTime) => React.ReactNode;
}

export type CountDownCurrentTime = CurrentTime;

export type CountDownInstance = {
  start: () => void;
  pause: () => void;
  reset: () => void;
};
