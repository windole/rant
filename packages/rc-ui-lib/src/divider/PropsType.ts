import React from 'react';
import { BaseTypeProps } from '../utils';

export type DividerContentPosition = 'left' | 'center' | 'right';
export interface DividerProps extends BaseTypeProps, React.HTMLAttributes<HTMLDivElement> {
  /** 是否使用虚线	 */
  dashed?: boolean;
  /**
   * 是否使用 0.5px 线
   * @default true
   */
  hairline?: boolean;
  /**
   * 内容位置
   * @default 'center'
   */
  contentPosition?: DividerContentPosition;
}
