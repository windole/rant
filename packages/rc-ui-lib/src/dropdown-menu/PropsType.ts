import React, { ReactNode } from 'react';
import { BaseTypeProps, TeleportType } from '../utils';

type DirectionTypes = 'up' | 'down';

export type DropdownItemOption = {
  /** 文案 */
  text?: string;
  /** 标识符 */
  value?: string | number;
  /** 左侧图标名称或图片链接 */
  icon?: string | ReactNode;
};

export interface DropdownMenuProps extends BaseTypeProps {
  /** 菜单标题和选项的选中态颜色 */
  activeColor?: string;
  /** 菜单展开方向 */
  direction?: DirectionTypes;
  /** 菜单栏 z-index 层级 */
  zIndex?: string | number;
  /** 动画时长，单位秒 */
  duration?: number | string;
  /** 是否显示遮罩层 */
  overlay?: boolean;
  /** 是否在点击遮罩层后关闭菜单 */
  closeOnClickOverlay?: boolean;
  /** 是否在点击外部元素后关闭菜单 */
  closeOnClickOutside?: boolean;
  /** 子元素 */
  children?: React.ReactNode;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** @private */
  shouldRenderTitle?: boolean;
}

export interface DropdownItemProps extends BaseTypeProps {
  /** 标志符 */
  name?: string;
  /** 菜单项标题 */
  title?: string;
  /** 默认值 */
  defaultValue?: DropdownItemOption;
  /** 当前选中项对应的 value */
  value?: DropdownItemOption;
  /** 占位文本 */
  placeholder?: string;
  /** 选项数组 */
  options?: DropdownItemOption[];
  /** 是否禁用菜单 */
  disabled?: boolean;
  /** 标题额外类名 */
  titleClass?: string;
  /** 子元素 */
  children?: React.ReactNode;
  /** @private */
  showPopup?: boolean;
  /** 指定挂载的节点 */
  teleport?: TeleportType;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** 打开菜单栏时触发 */
  onOpen?: () => void;
  /** 打开菜单栏且动画结束后触发 */
  onOpened?: () => void;
  /** 关闭菜单栏时触发 */
  onClose?: () => void;
  /** 关闭菜单栏且动画结束后触发 */
  onClosed?: () => void;
}

export interface DropdownMenuContextProps {
  props?: React.PropsWithChildren<DropdownMenuProps>;
  value?: Record<string, number | string>;
  offset?: number | string;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (obj) => void;
  renderTitle?: (ins: DropdownItemInstance) => void;
  close?: () => void;
}

export interface DropdownMenuInstance {
  close: () => void;
}
export interface DropdownItemInstance {
  /** 切换菜单展示状态，传 true 为显示，false 为隐藏，不传参为取反	 */
  toggle: (
    show?: boolean,
    options?: {
      immediate?: boolean;
    },
  ) => void;
  /** @private */
  state: {
    showPopup: boolean;
    transition: boolean;
    showWrapper: boolean;
  };
  /** @private */
  renderTitle: (value) => string | ReactNode;
}
