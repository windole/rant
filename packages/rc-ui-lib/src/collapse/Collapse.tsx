/* eslint-disable no-console */
import React, { ReactElement, useContext, useRef, useState } from 'react';
import classnames from 'classnames';
import CollapseContext from './CollapseContext';
import { CollapseProps } from './PropsType';
import { BORDER_TOP_BOTTOM } from '../utils/constant';
import { useUpdateEffect } from '../hooks';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';

function validateModelValue(
  modelValue: string | number | Array<string | number>,
  accordion: boolean,
) {
  if (accordion && Array.isArray(modelValue)) {
    console.error('[rc-ui-lib] Collapse: "value" should not be Array in accordion mode');
    return false;
  }
  if (!accordion && !Array.isArray(modelValue)) {
    console.error('[rc-ui-lib] Collapse: "value" should be Array in non-accordion mode');
    return false;
  }
  return true;
}

const Collapse: React.FC<CollapseProps> = (props) => {
  const { prefixCls, createNamespace } = useContext(ConfigProviderContext);
  const [bem] = createNamespace('collapse', prefixCls);

  const innerEffect = useRef(false);

  const [modelValue, setModelValue] = useState(() => props.value ?? props.initValue);

  const updateName = (name: number | string | Array<number | string>) => {
    innerEffect.current = true;
    setModelValue(name);
    props.onChange?.(name);
  };

  const toggle = (name, isExpanded: boolean) => {
    const { accordion } = props;
    if (accordion) {
      updateName(name === modelValue ? '' : name);
    } else if (isExpanded) {
      updateName((modelValue as Array<number | string>).concat(name));
    } else {
      updateName(
        (modelValue as Array<number | string>).filter((activeName) => activeName !== name),
      );
    }
  };

  const isExpanded = (name: string | number): boolean => {
    const { accordion } = props;
    if (process.env.NODE_ENV !== 'production' && !validateModelValue(modelValue, accordion)) {
      return false;
    }
    return accordion ? modelValue === name : (modelValue as Array<number | string>).includes(name);
  };

  useUpdateEffect(() => {
    if (innerEffect.current) {
      innerEffect.current = false;
      return;
    }
    setModelValue(props.value);
  }, [props.value]);

  return (
    <CollapseContext.Provider value={{ isExpanded, toggle }}>
      <div className={classnames(bem(), { [BORDER_TOP_BOTTOM]: props.border })}>
        {React.Children.toArray(props.children)
          .filter(Boolean)
          .map((child: ReactElement, index: number) =>
            React.cloneElement(child, {
              index,
            }),
          )}
      </div>
    </CollapseContext.Provider>
  );
};

Collapse.defaultProps = {
  border: true,
  initValue: [],
};

export default Collapse;
