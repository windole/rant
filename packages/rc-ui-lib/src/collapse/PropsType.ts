import { ReactNode } from 'react';
import { CellProps } from '../cell/PropsType';
import { BaseTypeProps } from '../utils';

export interface CollapseProps extends BaseTypeProps {
  /** 是否开启手风琴模式	 */
  accordion?: boolean;
  /** 默认打开的面板name */
  initValue?: string | number | Array<number | string>;
  /** 是否显示外边框	 */
  border?: boolean;
  /** 当前展开面板的 name	 */
  value?: number | string | Array<number | string>;
  /** 子组件	 */
  children?: ReactNode;
  /** 切换面板时触发	*/
  onChange?: (activeNames: number | string | Array<number | string>) => void;
}

export interface CollapseItemProps extends CellProps {
  /** 唯一标识符，默认为索引值	 */
  name?: number | string;
  /** 是否禁用面板	 */
  disabled?: boolean;
  /** 是否为只读状态，只读状态下无法操作面板	 */
  readonly?: boolean;
  /** @private */
  index?: number;
  /** 子组件	 */
  children?: ReactNode;
  // onClick: () => void;
}

export type CollapseItemInstance = {
  toggle: (expand: boolean) => void;
};
