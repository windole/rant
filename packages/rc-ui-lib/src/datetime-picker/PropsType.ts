import { PickerInstance, PickerProps } from '../picker/PropsType';

export type DatetimePickerColumnType = 'year' | 'month' | 'day' | 'hour' | 'minute';

export type DatetimePickerType =
  | 'date'
  | 'time'
  | 'datetime'
  | 'datehour'
  | 'month-day'
  | 'year-month';

export interface SharedProps extends Omit<PickerProps, 'onConfirm' | 'onChange' | 'columns'> {
  value?: string | Date;
  /** 选项过滤函数	 */
  filter?: (type: string, values: string[]) => string[];
  /**
   * 自定义列排序数组,
   * 子项可选值为: year、month、day、hour、minute
   */
  columnsOrder?: string[];
  /** 选项格式化函数	*/
  formatter?: (type: string, value: string) => string;
  /** 点击完成按钮时触发的事件	 */
  onConfirm?: (value: Date) => void;
  /** 当值变化时触发的事件	 */
  onChange?: (value: Date) => void;
  onCancel?: () => void;
}

export interface DatePickerProps extends SharedProps {
  /** 时间类型 */
  type?: DatetimePickerType;
  /** 可选的最小时间，精确到分钟	 */
  minDate?: Date;
  /** 可选的最大时间，精确到分钟	 */
  maxDate?: Date;
}

export interface TimePickerProps extends SharedProps {
  /** 可选的最小小时	 */
  minHour?: number | string;
  /** 可选的最大小时	 */
  maxHour?: number | string;
  /** 可选的最小分钟	 */
  minMinute?: number | string;
  /** 可选的最大分钟	 */
  maxMinute?: number | string;
}

export type DateTimePickerProps = DatePickerProps & TimePickerProps;

export type DateTimePickerInstance = {
  getPicker: () => PickerInstance;
};
