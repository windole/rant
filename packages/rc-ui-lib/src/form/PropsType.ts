import React from 'react';
import type { FormProps as RcFormProps, FormInstance as RcFormInstance } from 'rc-field-form';
import type { FieldProps as RcFieldProps } from 'rc-field-form/lib/Field';
import type { Meta } from 'rc-field-form/lib/interface';
import type { BaseTypeProps } from '../utils';
import type { CellGroupProps } from '../cell/PropsType';
import { FieldProps } from '../field/PropsType';

export type FormLayout = 'vertical' | 'horizontal';

export type FormValidateTrigger = 'onBlur' | 'onChange' | 'onSubmit';
type OmitFromCellGroupProps = Omit<CellGroupProps, 'title'>;
export interface FormProps extends RcFormProps, OmitFromCellGroupProps, BaseTypeProps {
  /** 表单布局 */
  layout?: FormLayout;
  /** 是否显示 label 后面的冒号 */
  colon?: boolean;
  /** 表单底部内容	 */
  footer?: React.ReactNode;
  /** 是否显示验证信息 */
  showValidateMessage?: boolean;
  /** 子元素 */
  children?: React.ReactNode;
  validateTrigger?: FormValidateTrigger;
}

export type RenderChildren<Values = unknown> = (form: RcFormInstance<Values>) => React.ReactNode;
type ChildrenType<Values = unknown> = RenderChildren<Values> | React.ReactNode;

export type FormInstance = RcFormInstance;

export type MemoInputProps = {
  value: unknown;
  update: number;
  children: React.ReactNode;
} & Record<string, unknown>;

export type FormItemProps = RcFieldProps &
  Pick<FormProps, 'showValidateMessage' | 'validateTrigger'> &
  Pick<
    FieldProps,
    | 'style'
    | 'className'
    | 'tooltip'
    | 'intro'
    | 'colon'
    | 'labelWidth'
    | 'labelAlign'
    | 'labelClass'
    | 'onClick'
  > & {
    /** 输入框左侧文本 */
    label?: string;
    /** 必填项 */
    required?: boolean;
    noStyle?: boolean;
    /** 是否禁用表单项 */
    disabled?: boolean;
    /** 自定义item，此时不会渲染内置的field */
    customField?: boolean;
    /** 子元素 */
    children?: ChildrenType;
  };

export type FormItemLayoutProps = Pick<
  FormItemProps,
  | 'required'
  | 'disabled'
  | 'label'
  | 'style'
  | 'className'
  | 'tooltip'
  | 'intro'
  | 'colon'
  | 'labelWidth'
  | 'labelAlign'
  | 'labelClass'
  | 'showValidateMessage'
  | 'validateTrigger'
> & {
  onClick?: (e?: React.MouseEvent) => void;
  htmlFor?: string;
  meta?: Meta;
  layout?: FormLayout;
  /** @private */
  isFieldChildren?: boolean;
};
