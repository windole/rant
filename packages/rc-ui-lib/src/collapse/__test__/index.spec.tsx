import * as React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import toJson from 'enzyme-to-json';
import Cell from '../../cell';
import Collapse from '..';
import { CollapseItemInstance } from '../PropsType';

describe('Collapse', () => {
  let wrapper;
  let spyConsole: jest.SpyInstance;
  beforeEach(() => {
    spyConsole = jest.spyOn(console, 'error');
    spyConsole.mockImplementation(() => {
      return null;
    });
  });

  afterEach(() => {
    wrapper.unmount();
    spyConsole.mockRestore();
  });

  // it('should update value when title is clicked', async () => {
  //   const onChange = jest.fn();
  //   wrapper = mount(
  //     <Collapse value={[]} initValue="1" onChange={onChange}>
  //       <Collapse.Item title="标题1" name="first">
  //         内容
  //       </Collapse.Item>
  //       <Collapse.Item title="标题2" name="2">
  //         内容
  //       </Collapse.Item>
  //     </Collapse>,
  //   );
  //   const title = wrapper.find(Cell).at(0).children('.rc-collapse-item__title');
  //   await title.simulate('click');
  //   expect(onChange).toHaveBeenCalledWith(['first']);
  // });

  it('should update value when title is clicked in accordion mode', async () => {
    const onChange = jest.fn();
    wrapper = mount(
      <Collapse value="" initValue="1" accordion onChange={onChange}>
        <Collapse.Item title="标题1" name="first">
          内容
        </Collapse.Item>
        <Collapse.Item title="标题2" name="2">
          内容
        </Collapse.Item>
      </Collapse>,
    );
    const title = wrapper.find(Cell).at(0).children('.rc-collapse-item__title');
    await title.simulate('click');
    expect(onChange).toHaveBeenCalledWith('first');
  });

  it('should render CollapseItem correctly', () => {
    wrapper = mount(
      <Collapse value={[]}>
        <Collapse.Item
          title="标题1"
          name="first"
          icon="shop-o"
          value="this is value"
          rightIcon="question-o"
        >
          内容
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not render border when border prop is false', () => {
    wrapper = mount(<Collapse border={false} />);
    expect(wrapper.find('.rc-hairline--top-bottom').exists()).toBeFalsy();
  });

  it('should toggle collapse after calling the toggle method', async () => {
    const CollapseItemRef = React.createRef<CollapseItemInstance>();
    const onChange = jest.fn();

    const toggle = (value?: boolean) => {
      CollapseItemRef.current.toggle(value);
    };

    wrapper = mount(
      <Collapse initValue={[]} onChange={onChange}>
        <Collapse.Item title="标题1" name="first" ref={CollapseItemRef}>
          内容
        </Collapse.Item>
      </Collapse>,
    );

    await act(async () => {
      toggle(true);
    });

    expect(onChange).toHaveBeenCalledWith(['first']);

    await act(async () => {
      toggle(false);
    });

    expect(onChange).toHaveBeenCalledWith([]);
  });

  it('should toggle collapse after calling the toggle method in accordion mode', async () => {
    const CollapseItemRef = React.createRef<CollapseItemInstance>();
    const onChange = jest.fn();

    const toggle = (value?: boolean) => {
      CollapseItemRef.current.toggle(value);
    };

    wrapper = mount(
      <Collapse initValue="" accordion onChange={onChange}>
        <Collapse.Item title="标题1" name="first" ref={CollapseItemRef}>
          内容
        </Collapse.Item>
      </Collapse>,
    );

    await act(async () => {
      toggle(true);
    });

    expect(onChange).toHaveBeenCalledWith('first');

    await act(async () => {
      toggle(false);
    });

    expect(onChange).toHaveBeenCalledWith('');
  });

  it('should be readonly when using readonly prop', async () => {
    const onChange = jest.fn();
    wrapper = mount(
      <Collapse initValue="" onChange={onChange}>
        <Collapse.Item title="标题1" name="first" readonly>
          内容
        </Collapse.Item>
      </Collapse>,
    );
    const title = wrapper.find(Cell).children('.rc-collapse-item__title');
    await title.simulate('click');
    expect(onChange).not.toHaveBeenCalled();
    expect(wrapper.find('i').exists()).toBeFalsy();
    expect(spyConsole).toHaveBeenCalledWith(
      '[rc-ui-lib] Collapse: "value" should be Array in non-accordion mode',
    );
  });

  it('should be readonly when using disabled prop', async () => {
    const onChange = jest.fn();
    wrapper = mount(
      <Collapse initValue="" onChange={onChange}>
        <Collapse.Item title="标题1" name="first" disabled>
          内容
        </Collapse.Item>
      </Collapse>,
    );
    const title = wrapper.find(Cell).children('.rc-collapse-item__title');
    await title.simulate('click');
    expect(onChange).not.toHaveBeenCalled();
    expect(wrapper.find('.rc-collapse-item__title--disabled').exists()).toBeTruthy();
    expect(spyConsole).toHaveBeenCalledWith(
      '[rc-ui-lib] Collapse: "value" should be Array in non-accordion mode',
    );
  });

  it('when value is change', async () => {
    wrapper = mount(
      <Collapse initValue="" accordion value="">
        <Collapse.Item title="标题1" name="first">
          内容
        </Collapse.Item>
      </Collapse>,
    );

    const title = wrapper.find(Cell).children('.rc-collapse-item__title');
    title.simulate('click');
    await wrapper.setProps({ value: 'first' });
    expect(wrapper.props().value).toEqual('first');
  });
});
