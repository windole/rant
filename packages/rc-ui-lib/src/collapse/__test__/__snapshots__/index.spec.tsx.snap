// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should render CollapseItem correctly 1`] = `
<Collapse
  border={true}
  initValue={Array []}
  value={Array []}
>
  <div
    className="rc-collapse rc-hairline--top-bottom"
  >
    <CollapseItem
      border={true}
      icon="shop-o"
      index={0}
      isLink={true}
      key=".0"
      name="first"
      rightIcon="question-o"
      title="标题1"
      value="this is value"
    >
      <div
        className="rc-collapse-item"
      >
        <Cell
          aria-expanded="false"
          className="rc-collapse-item__title"
          icon="shop-o"
          index={0}
          isLink={true}
          name="first"
          onClick={[Function]}
          rightIcon="question-o"
          title="标题1"
          value="this is value"
        >
          <div
            className="rc-cell rc-cell--clickable rc-collapse-item__title"
            onClick={[Function]}
            role="button"
          >
            <Icon
              className="rc-cell__left-icon"
              name="shop-o"
              tag="i"
            >
              <Badge
                className="rc-cell__left-icon van-icon van-icon-shop-o"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-shop-o"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
            <div
              className="rc-cell__title"
            >
              标题1
            </div>
            <div
              className="rc-cell__value"
            >
              <span>
                this is value
              </span>
            </div>
            question-o
          </div>
        </Cell>
      </div>
    </CollapseItem>
  </div>
</Collapse>
`;
